src/m365/entra/commands/user/user-list.ts

```diff
@@ -91,38 +120,24 @@ class EntraUserListCommand extends GraphCommand {
     }
   }
 
-  private getFilter(options: Options): string {
-    const filters: any = {};
-    const excludeOptions: string[] = [
-      'type',
-      'properties',
-      'p',
-      'd',
-      'debug',
-      'verbose',
-      'output',
-      'o',
-      'query',
-      '_'
-    ];
+  private getFilter(options: Options, type?: string): string {
 
+    const filters: any = {};
     Object.keys(options).forEach(key => {
-      if (excludeOptions.indexOf(key) === -1) {
-        if (typeof options[key] === 'boolean') {
-          throw `Specify value for the ${key} property`;
-        }
-
-        filters[key] = formatting.encodeQueryParameter(options[key].toString());
+      if (typeof options[key] === 'boolean') {
+        throw `Specify value for the ${key} property`;
       }
+
+      filters[key] = formatting.encodeQueryParameter(options[key].toString());
     });
 
     let filter: string = Object.keys(filters).map(key => `startsWith(${key}, '${filters[key]}')`).join(' and ');
     if (filter.length > 0) {
       filter = `$filter=${filter}`;
     }
 
-    if (options.type) {
-      filter += filter.length > 0 ? ` and userType eq '${options.type}'` : `$filter=userType eq '${options.type}'`;
+    if (type) {
+      filter += filter.length > 0 ? ` and userType eq '${type}'` : `$filter=userType eq '${type}'`;
```

Comment:
Now we're checking twice if there are any filters. I think it's a lot clearer if we throw all filters in a single array and use that array as a filter string.